src/Interpreters/Context.cpp

```diff
@@ -2478,6 +2478,7 @@ void Context::setFormatSchemaPath(const String & path)
 
 Context::SampleBlockCache & Context::getSampleBlockCache() const
```

Comment:
What will happen if we will simply remove "sample block cache"?

Reply:
Queries like [this one](https://github.com/ClickHouse/ClickHouse/blob/97f2a2213e754ba25dabba4bc8ddf507cd67660c/tests/queries/0_stateless/00632_get_sample_block_cache.sql#L63-L175) will significantly slow down, see #2313

Reply:
We need to check if it is still the case.

Reply:
Yes, I've checked it, test fails without sample block cache

Reply:
Maybe we should create query context in `MergeTreeDataMergerMutator`, so mutations will be able to use sample block cache too, but I don't think that someone will run mutations with huge number of subqueries